modules/new_profile/jsx/NewProfileIndex.js

```diff
@@ -0,0 +1,244 @@
+import Panel from 'Panel';
+import Loader from 'Loader';
+/**
+ * New Profile Form
+ *
+ * Create a new profile form
+ *
+ * @author  Shen Wang
+ * @version 1.0.0
+ * */
+class NewProfileIndex extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: {},
+      configData: {},
+      formData: {},
+      newData: {},
+      isLoaded: false,
+      isCreated: false,
+      errMessage: '',
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.setFormData = this.setFormData.bind(this);
+    this.fetchData = this.fetchData.bind(this);
+  }
+
+  componentDidMount() {
+    this.fetchData()
+      .then(() => this.setState({isLoaded: true}));
+  }
+
+  /**
+   * Retrieve data from the provided URL and save it in state
+   *
+   * @return {object}
+   */
+  fetchData() {
+    return fetch(this.props.dataURL,
+      {credentials: 'same-origin'})
+      .then((resp) => resp.json())
+      .then((data) => this.setState({configData: data}))
+      .catch((error) => {
+        this.setState({error: true});
+      });
+  }
+  /**
+   * It checks the date of birth and Expected Date of Confinement,
+   * the date fields must match.
+   * If match, this function will return true.
+   *
+   * @return {boolean}
+   */
+  validateMatchDate() {
+    const data = this.state.formData;
+    let dateMatch = false;
+    if (data.dateTaken == data.dateTakenConfirm) {
+      dateMatch = true;
+    }
+    let edcMatch = true;
+    if (this.state.configData['edc'] === 'true' &&
+         data.edcDateTaken !== data.edcDateTakenConfirm
+       ) {
+      edcMatch = false;
+    }
+    return dateMatch && edcMatch;
+  }
+
+  /**
+   * Handles form submission
+   *
+   * @param {event} e - Form submition event
+   */
+  handleSubmit(e) {
+    e.preventDefault();
+    const match = this.validateMatchDate();
+    if (!match) {
+      this.setState({
+        errMessage: 'Date of Birth or EDC fields must match',
+        isCreated: false,
+      });
+    } else {
+      let formData = this.state.formData;
+      let formObject = new FormData();
+      for (let key in formData) {
+        if (formData[key] !== '') {
+          formObject.append(key, formData[key]);
+        }
+      }
+      fetch(this.props.submitURL, {
+        method: 'POST',
+        cache: 'no-cache',
+        credentials: 'same-origin',
+        body: formObject,
+        })
+      .then((resp) => resp.json())
+      .then((data) => {
+        this.setState({newData: data});
+        this.setState({isCreated: true});
+       });
+    }
+  }
+
+  /**
+   * Set the form data based on state values of child elements/components
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding form element
+   */
+  setFormData(formElement, value) {
+    let formData = JSON.parse(JSON.stringify(this.state.formData));
```

Comment:
Hi @kongtiaowang a cleaner approach compared to the `JSON.parse(JSON.stringify(this.state.formData));` is instead doing something like this: `let formData = Object.assign({}, this.state.formData);`